b2g/components/RemoteControlService.jsm

```diff
@@ -115,6 +128,26 @@ this.RemoteControlService = {
     }
   },
 
+  // SystemAppProxy event listener
+  handleEvent: function(evt) {
+    if (evt.type !== "mozContentEvent") {
+      return;
+    }
+
+    let detail = evt.detail;
+    if (!detail) {
+      return;
+    }
+
+    switch (detail.type) {
+      case "control-mode-changed":
+        // Use mozContentEvent to receive control mode of current app from System App
+        // command.js use "isCursorMode" to determine what kind event should dispatch to app
+        this._setSharedState("isCursorMode", detail.detail.cursor.toString());
```

Comment:
`isCursorMode` is read-only to sandbox, you should expose a separate getter function instead of using shared state.